src/Extend/Settings.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Api\Serializer\AbstractSerializer;
+use Flarum\Api\Serializer\ForumSerializer;
+use Flarum\Extension\Extension;
+use Flarum\Foundation\ContainerUtil;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Container\Container;
+
+class Settings implements ExtenderInterface
+{
+    private $settings = [];
+    private $defaults = [];
+    private $modifiers = [];
+
+    /**
+     * Serialize a setting value to the ForumSerializer attributes.
+     *
+     * @param string $attributeName: The attribute name to be used in the ForumSerializer attributes array.
+     * @param string $key: The key of the setting.
+     * @return $this
+     */
+    public function serializeToForum(string $attributeName, string $key)
+    {
+        $this->settings[$key] = $attributeName;
+
+        return $this;
+    }
+
+    /**
+     * Set a default value for a setting selected for serialization.
+     *
+     * @param string $key
+     * @param $value
+     * @return $this
+     */
+    public function default(string $key, $value)
+    {
+        $this->defaults[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Modify a setting's value.
+     *
+     * @param string $key
+     * @param $callback
+     * @return $this
+     */
+    public function modifier(string $key, $callback)
```

Comment:
What's the use case for this?

Reply:
There are some extensions that modify the value, for instance json decode it or cast it to a specific type. If we don't provide a way to do that, they'll just end up directly using the ApiSerializer extender and manually add them as attributes.

Would we want to also have this not only done while serializing, but in the repository level ? including it in the manager class we're talking about in the above comment. It does sound like a good idea to me.

Reply:
I'm still unsure about this one. For general purpose usage, I think providing this layer could be overengineering settings, at least right now. For serialization, the most I'd do is a optional callback that takes the setting as a third argument to `serializeToForum`, which, if provided, could perform some simple modification like you mentioned

Reply:
Alright let's keep it simple then